Anime example app initial commit

Potatomonsta opened the pull request:
AniWatch example app

<!--
  Thanks for contributing!

  Provide a description of your changes below and a general summary in the title

  Please look at the following checklist to ensure that your PR can be accepted quickly:
-->

## Description

Added Anime example app AniWatch

## Type of Change

<!--- Put an `x` in all the boxes that apply: -->

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code refactor
- [ ] Build configuration change
- [ ] Documentation
- [ ] Chore

Potatomonsta committed:
Anime example app initial commit

AniWatch example app

divyanshub024 commented on examples/ani_watch/README.md:
```diff
@@ -0,0 +1,16 @@
+# AniWatch
+
+A new Flutter project.
```
Can we add some proper documentation?

Potatomonsta replied:
added some documentation here

divyanshub024 commented on examples/ani_watch/assets/images/image.png:
Do we need this image?

Potatomonsta replied:
removed

divyanshub024 commented on examples/ani_watch/assets/images/jjk.jpg:
Do we need this image?

Potatomonsta replied:
removed

divyanshub024 commented on examples/ani_watch/assets/jsons/screens/onboarding_screen.json:
```diff
@@ -0,0 +1,103 @@
+{
+    "type": "scaffold",
+    "body": {
+        "type": "stack",
```
Do we need stac widget here? Can we replace it with column?

Potatomonsta replied:
yeah, since the bottom panel is placed on top of the image, in the future there  will  also be a a carousel indicator on top of the image

divyanshub024 replied:
Let's remove it now. If needed, we'll add it in future.

Potatomonsta replied:
its needed right now too, for the image to align properly (crop vertically at bottom) since we do not have fittedBox yet.

divyanshub024 commented on examples/ani_watch/assets/jsons/screens/onboarding_screen.json:
```diff
@@ -0,0 +1,103 @@
+                "imageType": "asset",
+                "src": "assets/images/kny.png",
+                "width": 10000,
+                "height": 10000,
```
Use `maxFinite` instead of `10000`.

Potatomonsta replied:
done

divyanshub024 commented on examples/ani_watch/assets/jsons/screens/onboarding_screen.json:
```diff
@@ -0,0 +1,103 @@
+                                    "height": 10
+                                },
+                                {
+                                    "type": "text",
```
Do we need to put text in the children? Can we use it directly in the data?

Potatomonsta replied:
rewrote to use copyWithStyle

divyanshub024 commented on examples/ani_watch/lib/screens/navigation_view.dart:
```diff
@@ -0,0 +1,104 @@
+      "padding": 0,
+      "child": {
+        "type": "clipRRect",
+        "child": {
+          "type": "stack",
+          "fit": "expand",
+          "children": [
+            {
+              "type": "backdropFilter",
+              "filter": {"type": "blur", "sigmaX": 20.0, "sigmaY": 20.0},
+            },
```
Can we remove the stack and backdrop filter?

Potatomonsta replied:
pushed a fix to make it hopefully work for iOS

divyanshub024 replied:
awesome

divyanshub024 commented on examples/ani_watch/lib/screens/home.dart:
```diff
@@ -0,0 +1,322 @@
+        sortType: AniListAPIUtils.sortTrendingDesc,
+      ),
+      // {
+      //   "type": "sizedBox",
```
Can we remove the comments?

Potatomonsta replied:
These are additional sections that are temporarily commented out to reduce API calls, we might enable them later

divyanshub024 replied:
Let's remove it for better code readability. We can add it in future if needed.

Potatomonsta replied:
removed

divyanshub024 commented on examples/ani_watch/fonts/Figtree-Italic-VariableFont_wght.ttf:
Can we remove the font file which are not needed?

Potatomonsta replied:
should we remove italics? its part of the font and we might use italics later, also could be part of anime description(not handled atm, but something to do in the future)

divyanshub024 replied:
Let's remove it. We can add it in future if needed.

Potatomonsta replied:
removed

divyanshub024 commented on examples/ani_watch/lib/widgets/movie_carousel/movie_carousel.dart:
Can we please use the correct file name?

Potatomonsta replied:
done, renamed to anime_home

divyanshub024 reviewed (changes requested):
Thanks for the amazing PR @Potatomonsta 🤩 
I have some suggestions please check them out!!

Potatomonsta committed:
Bugfixes and cleanup

- Removed unused images
- Minor changes to onboarding to make it more up to date with Stac changes
- Renamed movie_carousel to anime_home_carousel
- Episode list is not reversed on details screen now
- Bottom nav height adjusted for iOS
- Updated Readme.md

Potatomonsta committed:
Minor changes

- removed commented code in home
- removed Figtree italic files
